tests/apidoc_login_test.go

```diff
@@ -41,34 +42,76 @@ func TestLogin(t *testing.T) {
 		// sytest: POST /login can log in as a user
 		t.Run("POST /login can login as user", func(t *testing.T) {
 			t.Parallel()
-			res := unauthedClient.MustDo(t, "POST", []string{"_matrix", "client", "r0", "register"}, json.RawMessage(`{
-				"auth": {
-					"type": "m.login.dummy"
+			err := createDummyUser(t, unauthedClient, "login_test_user")
+			if err != nil {
+				t.Fatalf("unable to create dummy user %v", err)
+			}
+			res := unauthedClient.MustDo(t, "POST", []string{"_matrix", "client", "r0", "login"}, json.RawMessage(`{
+				"type": "m.login.password",
+				"identifier": {
+					"type": "m.id.user",
+					"user": "login_test_user"
 				},
-				"username": "post-login-user",
 				"password": "superuser"
 			}`))
+
 			must.MatchResponse(t, res, match.HTTPResponse{
 				JSON: []match.JSON{
 					match.JSONKeyTypeEqual("access_token", gjson.String),
-					match.JSONKeyTypeEqual("user_id", gjson.String),
+					match.JSONKeyEqual("home_server", "hs1"),
 				},
 			})
-			res = unauthedClient.MustDo(t, "POST", []string{"_matrix", "client", "r0", "login"}, json.RawMessage(`{
+		})
+		// sytest: POST /login returns the same device_id as that in the request
+		t.Run("POST /login returns the same device_id as that in the request", func(t *testing.T) {
+			t.Parallel()
+			deviceId := "test_device_id"
+			err := createDummyUser(t, unauthedClient, "device_id_test_user")
+			if err != nil {
+				t.Fatalf("unable to create dummy user %v", err)
+			}
+			res := unauthedClient.MustDo(t, "POST", []string{"_matrix", "client", "r0", "login"}, json.RawMessage(`{
 				"type": "m.login.password",
 				"identifier": {
 					"type": "m.id.user",
-					"user": "post-login-user"
+					"user": "device_id_test_user"
 				},
-				"password": "superuser"
+				"password": "superuser",
+				"device_id": "`+deviceId+`"
 			}`))
 
 			must.MatchResponse(t, res, match.HTTPResponse{
 				JSON: []match.JSON{
 					match.JSONKeyTypeEqual("access_token", gjson.String),
-					match.JSONKeyEqual("home_server", "hs1"),
+					match.JSONKeyEqual("device_id", deviceId),
 				},
 			})
 		})
 	})
 }
+
+func createDummyUser(t *testing.T, unauthedClient *client.CSAPI, userId string) error {
```

Comment:
When specifying acronyms, please use upper case letters e.g `userID` not `userId`. This is what many Go developers expect to see, and some linters will fail code without this.